Clean up readme, link to wiki

crxporter opened the pull request:
I got stuck on the JSON characteristic properties so I decided to make some updates to readme.

Much of the information in README was very old. I don't think updates from the now 2-year-old node-red-homekit need to be front and center anymore.

Also added some explanation of parent/linked services.

Shaquu commented:
So let's begin some disscusion :)
First things first I agree we need to remake the README completly.

As you removed _story_ part from README could you make a page for it in the wiki?
It's important for me to keep people that created that module rewarded even with only a mention in the page.

Another thing is that I can't agree that Linked service create sub accessory. It's just not right for me to call it that way according to HAP docs.
![accessories](https://user-images.githubusercontent.com/2881159/54194668-11398200-44bd-11e9-8de1-4c00787ca6a6.png)

So Parent and Linked Services create Accessory as a whole even if you so 2 or more icons in Home.app

crxporter commented:
> As you removed story part from README could you make a page for it in the wiki?

I moved “why this fork” to the bottom as I think it’s important but I personally like to see getting started and install instructions first. 

> So Parent and Linked Services create Accessory as a whole even if you so 2 or more icons in Home.app

Good point. I’ll work on that wording this afternoon/evening.

Shaquu commented:
It looks great!

Here are my comments and ideas (you do not have to hold on to them of course :) )
1. [Missing answer in FAQ](https://github.com/node-red-contrib-homekit/node-red-contrib-homekit-bridged/blob/9bf018a582ae35369bea479fbd42c35e827e6837/README.md#the-same-command-gets-sent-over-and-over-how-do-i-stop-that)
2. Add Credits page to Wiki sidebar as currently we can only find it via README link
3. Also it would be good to add Wiki Services page link next to [Supported Types section](https://github.com/node-red-contrib-homekit/node-red-contrib-homekit-bridged/blob/9bf018a582ae35369bea479fbd42c35e827e6837/README.md#supported-types).
4. Maybe move "Based on node-red-contrib-homekit, but with support for bridged devices." to Credits? - according to [this](https://github.com/node-red-contrib-homekit/node-red-contrib-homekit-bridged/graphs/contributors/) we made more changes in the code then the length of original fork. So I think it's time to cut ourself as just a fork since we are the only (or on of the best) alternative in the market.
5. Add in intro that we can use acccessories not only in "Apple's Home app on iOS and Mac OS" but also on Watch OS
6. Add Topic description for Servics

That for now :)

crxporter commented:
I’m good to work on all but 6. 

Anyone have a description for Topic? I don’t use that and I’m not entirely sure what it’s there for.

Shaquu commented:
Okay. So question to the authors - @flic @emilingerslev

flic commented:
This is from the help text:

An optional property that can be configured in the node or, if left blank, can be set by **msg.topic**.
If _Filter on Topic_ is selected **msg.topic** of incoming messages must match the configured value for the message to be accepted. If _Filter on Topic_ is selected and no Topic is set on the node, then **msg.topic** must match the nodes Name.

...and I suggest maybe adding:

The Topic parameter can be used to filter incoming messages, making it possible to connect multiple Homekit services to, for example, one MQTT-in node and filter directly on the MQTT Topic. It can also be used to add additional metadata to the outgoing msg, making it possible to connect multiple Homekit services directly to an MQTT-out node or filter the flow in another way.

Feel free to change it as you like.

crxporter commented:
Thank's @flic! Great timing I was just working on that when I saw your comment.

Shaquu commented:
1. Doubled Topic under Service
2. What is rbe node? I really don't know. Do I?

crxporter commented:
1. Done-good catch. 
2. “Return by exception” is [built in](
https://flows.nodered.org/node/node-red-node-rbe)

Shaquu commented:
1. About HAP-Nodejs. What about replacing
 

> HAP-NodeJS -Project which uses an implementation of mdns to provide Bonjour / Avahi capability.

with

> HAP-NodeJS - Node.js implementation of the HomeKit Accessory Server.

2. We miss MDNS configuration from Bridge section. #44 
3. I wonder if it should be merged. For me all of this fits into Intro section.

> Node-RED nodes to simulate Apple HomeKit devices. The goal is to emulate native HomeKit devices as closely as possible. We rely on community support - please read throught the README for the basics then head over to the wiki page for details and examples. If you're still stuck please open an issue, we are glad to help.
> 
> Intro
> These nodes allow the creation of fully customizable accessories for use in Apple's Home app on iOS, Watch OS, and Mac OS. If you can get it in Node-RED, you can get it in HomeKit. The goal of the project is to create a platform where official HomeKit hardware can be emulated as closely as possible through node red.
4. Another thing is Prerequisites section. I wonder if we don't misslead people here - we don't want them to woory about HAP-Nodejs until they have a problem with it. Now it looks like they should read about HAPNodejs or something?
5. Raymond's user is RaymondMouthaan so maybe change this (still with link to profile):

> thanks to raymondmm (Raymond Mouthaan).
to
> thanks to Raymond Mouthaan
6. We need a mark that Bridge is indeed configuration node but only exist in Service node (or is part of Service configuration or something like this). To not make people worry about missing node in the sidebar.
7. Linked. each - there is a dot so Each (capital letter)?
8. Make Bridge and Parent Service with deeper indentation on the list as they are under Service Hierarchy
9. on the wiki.If  - space after dot
10. I am thinking about removing HAP-NodeJS from contributors list as it is obvious from the beginning of the readme that we are using this as our core

Cheers :)

Shaquu commented:
From homekit.html regardin mdns:
<li><strong>Custom MDNS Configuration</strong>: Check if you would like to use custom mdns configuration.</li>
        <ul>
            <li><strong>Multicast</strong>: Use udp multicasting. Optional. Default true.</li>
            <li><strong>Multicast Interface IP:</strong>: Explicitly specify a network interface. Optional. Defaults to all.</li>
            <li><strong>Port</strong>: Set the udp port. Optional. Default 5353.</li>
            <li><strong>Multicast Address IP</strong>: Set the udp ip. Optional. </li>
            <li><strong>TTL</strong>: Set the multicast ttl. Optional. </li>
            <li><strong>Loopback</strong>: Receive your own packets. Optional. Default true.</li>
            <li><strong>Reuse Address</strong>: Set the reuseAddr option when creating the socket. Optional. Default true.</li>
        </ul>

crxporter commented:
1. Removed in response to 4 below
2. Added.
3. Moved “intro” header up.
4. Removed - correct this isn’t a prerequisite, I don’t know who typed that part but it’s misleading. The prerequisite is potentially installing Bonjour / Avahi (in the install instruction)
5. I have no idea who put that link in. Fixed.
6. Addressed, take a look.
7. Aren’t you Polish? How is your English better than mine?
8. Done.
9. Again with the English.
10. Removed from up above, I vote leave it at the bottom only.

Shaquu commented:
How can you know I am from :poland: ? 
Regarding 10 - no need to vote. You rule here and I just propose.

Well I think we should merge. I can't read it more :smile:

crxporter commented:
Your profile gives it away! Has the little 🇵🇱 and everything...

Merge away. Someone else can change it later if they don’t like it. 

Thanks!

Shaquu commented:
@crxporter thank you so much for your contribution! I am happy to have you here.
